### PR TITLE
feat(create): adds new add deploy config command

### PR DESCRIPTION
**Pull request by cianmSAP**
https://github.com/SAP/open-ux-tools/issues/2128

- Adds command `add deploy-config` to create module, uses `getPrompts` from `@sap-ux/abap-deploy-config-inquirer`

### CODE REVIEW

File: packages/abap-deploy-config-inquirer/src/index.ts [hunk 1]
```diff
@@ -24,7 +26,7 @@ import { reconcileAnswers } from './utils';
  * @returns the prompts used to provide input for abap deploy config generation and a reference to the answers object which will be populated with the user's responses once `inquirer.prompt` returns
  */
 async function getPrompts(
-    promptOptions?: AbapDeployConfigPromptOptions,
+    promptOptions: AbapDeployConfigPromptOptions,
```
Comment (IainSAP) [inline packages/abap-deploy-config-inquirer/src/index.ts:29]: Prompt options should be eh optional and default behaviour should be usable?

File: packages/abap-deploy-config-inquirer/src/prompts/conditions.ts [hunk 2]
```diff
@@ -182,12 +182,15 @@ function defaultOrShowPackageQuestion(): boolean {
...
+ * @param useAutocomplete - useAutocomplete option from prompt options
  * @returns boolean
  */
-export function showPackageInputChoiceQuestion(): boolean {
-    // Only show the input choice (manual/search) when the autocomplete prompt is supported; CLI or YUI specific version
+export function showPackageInputChoiceQuestion(useAutocomplete = false): boolean {
+    // Only show the input choice (manual/search) when the autocomplete option is true, the prompt is supported; CLI or YUI specific version
     return Boolean(
-        (!PromptState.isYUI || isFeatureEnabled('enableAutocompleteUIPrompt')) && defaultOrShowPackageQuestion()
+        (!PromptState.isYUI || isFeatureEnabled('enableAutocompleteUIPrompt')) &&
```
Comment (IainSAP) [inline packages/abap-deploy-config-inquirer/src/prompts/conditions.ts:191]: Not clear why we are using feature toggle for an option?

**Review by IainSAP — COMMENTED** [review]
Some comments

File: packages/abap-deploy-config-inquirer/src/index.ts (hunk 1 above)
Comment (cianmSAP) [reply packages/abap-deploy-config-inquirer/src/index.ts]: Updated

File: packages/abap-deploy-config-inquirer/src/prompts/conditions.ts (hunk 2 above)
Comment (cianmSAP) [reply packages/abap-deploy-config-inquirer/src/prompts/conditions.ts]: Removed as discussed

File: packages/abap-deploy-config-inquirer/src/index.ts [hunk 3]
```diff
@@ -45,20 +47,29 @@ async function getPrompts(
...
+    adapter?: InquirerAdapter,
     logger?: Logger,
     isYUI = false
 ): Promise<AbapDeployConfigAnswers> {
     const abapDeployConfigPrompts = (await getPrompts(promptOptions, logger, isYUI)).prompts;
-    const answers = await adapter.prompt<AbapDeployConfigAnswersInternal>(abapDeployConfigPrompts);
+
+    const pm = adapter ? adapter.promptModule : inquirer;
+
+    if (pm && promptOptions?.useAutocomplete) {
```
[the remark this answers is not in the document]
Comment (cianmSAP) [reply packages/abap-deploy-config-inquirer/src/index.ts:66]: I've left this as a general option for now as discussed.

**Review by IainSAP — APPROVED** [review]
Thanks @cianmSAP 

- Comments addressed
- Tests added to cover
- Not manually tested

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
* changeset ok
* test coverage ok
* code is clear and adheres to coding standards
* local test successful

One request: please check your usages of the `as` operator, and consider `satisfies` instead. It is the better choice in `cli/add/deploy-config`

**Review by tobiasqueck — APPROVED** [review]
requested changes have been applied

**Review by longieirl — APPROVED** [review]
Changes look good.
Changeset is good.
Test coverage is good.
Once the main PR into tools-suite is merged I will consume this latest update and validate.

**Pull request merged**